activerecord/lib/active_record/migration.rb

```diff
@@ -138,7 +138,7 @@ class PendingMigrationError < MigrationError # :nodoc:
 
       if ActiveRecord.dump_schema_after_migration
         ActiveRecord::Tasks::DatabaseTasks.dump_schema(
-          ActiveRecord::Base.connection_db_config
+          (ActiveRecord.application_record_class || ActiveRecord::Base).connection_db_config
```

Comment:
Can you make this a method like how `def connection` is a method to clean this up the logic a bit?

Reply:
@eileencodes sorry for my late reply. Dunno why I didn't see the email notification. I think that will be a good improvement. Do you think `migration_base_class` or `migration_connection_class` would be short and descriptive enough?

Reply:
@eileencodes I went with `migrate_on_base_class` but happy to adjust if you like another name.

Reply:
@eileencodes did you get a chance to look at this since I made the change?